bin/node/src/system_monitor.rs

```diff
@@ -41,22 +41,33 @@ impl SystemMonitor {
         }
     }
 
-    /// Runs the system monitor loop on a separate thread.
-    pub fn run(mut self) {
-        std::thread::spawn(move || {
+    /// Runs a supervisor task that re-runs the system monitor if it panics.
+    pub fn run_with_supervisor(self) -> task::JoinHandle<()> {
+        task::spawn(async move {
             loop {
-                std::thread::sleep(self.monitor_interval);
-                if let Err(err) = self.log_system_metrics() {
+                if let Err(err) = self.clone().run().await {
```

Comment:
I think I'd add a sleep for some seconds here on the loop so that if for some reason it fails to restart, it does not retry as fast as possible.